services/ui-src/src/components/fields/DynamicField.tsx

```diff
@@ -59,6 +62,37 @@ export const DynamicField = ({ name, label, ...props }: Props) => {
     setDisplayValues(newDisplayValues);
   };
 
+  // if should autosave, submit field data to database on blur
+  const onBlurHandler = async (event: InputChangeEvent) => {
+    if (userIsStateUser || userIsStateRep) {
+      // check field data validity
+      const fieldDataIsValid = await form.trigger(event.target.name);
+      let data = displayValues;
+      // if valid, use; if not, reset to default
+      if (!fieldDataIsValid) {
+        data = displayValues.map((displayValue) => {
+          if (displayValue.id === event.target.id) displayValue.name = "";
+          return displayValue;
+        });
+      }
```

Comment:
Comment: Can probably improve readability here and avoid mapping through all items. The effect of this is really changing `data` from `displayValues` to `displayValues` with the blurred input value changed to `""`. So we could do something like I've included below. That said, we also don't need to do this right now because the only way for a dynamic field to be invalid is if the value is blank. So we're really just changing the value from `""` to `""`. Whether existing or modified, the `if` block is only necessary if we change the dynamic field functionality in the future, so I think we can just remove it for now, along with the field validity check and instead leave a note about why it's unnecessary.

```suggestion
      // check field data validity
      const fieldDataIsValid = await form.trigger(event.target.name);
      // prepare to use display values as field value for database
      let fieldInputValues = displayValues;
      // if field not valid, reset last blurred input to default value
      if (!fieldDataIsValid) {
        const blurredInputIndex = fieldInputValues.findIndex(
          (input: EntityShape) => input.id === event.target.id
        );
        fieldInputValues[blurredInputIndex].name = "";
      }
```

Reply:
Update: I removed this `if` block and added a comment in my commit here: https://github.com/CMSgov/mdct-mcr/pull/5336/commits/7bb0cbc4e95781c3724b99d5157f54c66166121a